pkg/component/runtime/runtime.go

```diff
@@ -132,15 +132,13 @@ func newComponentRuntimeState(m *Manager, logger *logger.Logger, monitor Monitor
 	})
 
 	// start the go-routine that watches for updates from the component
-	runner.Start(context.Background(), func(ctx context.Context) error {
```

Comment:
The runner returned here is immediately discarded so the context in the callback can never be cancelled, so I reverted to a bare goroutine to make the removal of the redundant case more clear